src/adagrams.js

```diff
@@ -1,6 +1,110 @@
+import { ENGINE_METHOD_DIGESTS } from "constants";
+
+const hashIntoObject = arrOfLetters => {
+  let object = {};
+  arrOfLetters.forEach(letter => {
+    object[letter] ? object[letter] += 1 : object[letter] = 1;
+  });
+  return object;
+}
+
 const Adagrams = {
   drawLetters() {
     // Implement this method for wave 1
+    let letterPool = 'AAAAAAAAABBCCDDDDEEEEEEEEEEEEFFGGGHHIIIIIIIIIJKLLLLMMNNNNNNOOOOOOOOPPQRRRRRRSSSSTTTTTTUUUUVVWWXYYZ'.split('');
+    let lettersInHand = [];
+    
+    for (let i = 0; i < 10; i += 1) {
+      const randomIndex = Math.floor(Math.random() * letterPool.length);
+      lettersInHand.push(letterPool[randomIndex]);
+      letterPool.splice(randomIndex, 1);
+    }
+    return lettersInHand;
+  },
+
+  usesAvailableLetters(input, lettersInHand) {
+    let lettersObject = hashIntoObject(lettersInHand);
+    for (let i in input) {
+      if (!lettersObject[input[i]] || lettersObject[input[i]] <= 0) return false;
+
+      lettersObject[input[i]] -= 1;
+    }
+    return true;
+  },
+
+  scoreWord(word) {
+    let score = 0;
+    if (word.length >= 7) {
+      score += 8;
+    }
+    word.toUpperCase().split('').forEach( (letter) => {
+      switch(letter) {
+        case 'A':
+        case 'E':
```

Comment:
I'm not a big fan of organizing the letter scores in a `switch` statement like this. It's not very DRY (you've had to type `case` 26 times), but more importantly the data is intertwined with the code that uses it, which means there's no easy way to reuse this information.

For example, what if the player had a hand full of tiles, and you wanted to be able to display the value of each of those letters? You would have to copy/paste this switch in that code, or at least extract it to a method.

A better approach would be to store the scores in a hash:

```javascript
const LETTER_VALUES = {
  a: 1,
  b: 3,
  // ...
}
```

then here you could use it like this:

```javascript
word.forEach((letter) => {
  const value = LETTER_VALUES[letter.toLowerCase()];
  if (value) {
    totalScore += letter.toLowerCase();
  } else {
    throw new Error(`Invalid character ${letter}`);
  }
}
```